src/axom/spin/tests/spin_bvh.cpp

```diff
@@ -177,34 +184,28 @@ void generate_aabbs_and_centroids(const mint::Mesh* mesh,
                 numerics::Matrix<double> & coords,
                 const IndexType* AXOM_UNUSED_PARAM(nodeIds)) {
       BoxType range;
-
-      PointType sum(0.0);
+      PointType sum;
 
       for(IndexType inode = 0; inode < NUM_NODES_PER_CELL; ++inode)
       {
         const double* node = coords.getColumn(inode);
 
         PointType coords;
-        for(int dim = 0; dim < NDIMS; dim++)
+        for(int dim = 0; dim < NDIMS; ++dim)
         {
           coords[dim] = node[dim];
           sum[dim] += node[dim];
         }
 
         range.addPoint(coords);
-      }  // END for all cells nodes
-      for(int dim = 0; dim < NDIMS; dim++)
-      {
-        sum[dim] /= NUM_NODES_PER_CELL;
       }
 
+      sum.array() /= NUM_NODES_PER_CELL;
+
       c[cellIdx] = range.getCentroid();
 
 #ifndef AXOM_DEVICE_CODE
-      for(int dim = 0; dim < NDIMS; dim++)
-      {
-        EXPECT_DOUBLE_EQ(c[cellIdx][dim], sum[dim]);
-      }
+      EXPECT_NEAR(primal::squared_distance(c[cellIdx], sum), 0., EPS);
```

Comment:
Note: This function is called with `FloatType = {float,double}` but the coordinates are `double`, so there is some loss in precision in some cases when taking averages.